keps/sig-cli/20191008-kubectl-events.md

```diff
@@ -0,0 +1,160 @@
+See custom columns [http://kubernetes.io/docs/user-guide/kubectl-overview/#custom-columns], golang template
+[http://golang.org/pkg/text/template/#pkg-overview] and jsonpath template
+[http://kubernetes.io/docs/user-guide/jsonpath].
+      --output-watch-events=false: Output watch event objects when --watch or --watch-only is used. Existing objects are
+output as initial ADDED events.
+      --raw='': Raw URI to request from the server.  Uses the transport specified by the kubeconfig file.
+  -l, --selector='': Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)
+      --server-print=true: If true, have the server return the appropriate table output. Supports extension APIs and CRDs.
+      --show-kind=false: If present, list the resource type for the requested object(s).
+      --show-labels=false: When printing, show all labels as the last column (default hide labels column)
+      --sort-by='': If non-empty, sort list types using this field specification.  The field specification is expressed
+as a JSONPath expression (e.g. '{.metadata.name}'). The field in the API resource specified by this JSONPath expression
+must be an integer or a string.
+  -w, --watch=false: After listing/getting the requested object, watch for changes. Uninitialized objects are excluded
+if no object name is provided.
+      --watch-only=false: Watch for changes to the requested object(s), without listing/getting first.
+--watch-event: watch for "create" or update events, filtering out deletes or any other event type:
+--watch-until: watch until the event exists
+--watch-until-exists: watch until an onbject exists
+--watch-for: watch for a particular object. 
+
+  ```
+  
+
+#### Long standing issues
+
+Following is a list of long standing issues for `events`
+
+- kubectl get events doesnt sort events by last seen time [kubernetes/kubernetes#29838](https://github.com/kubernetes/kubernetes/issues/29838)
+- Improve --watch behavior for events [kubernetes/kubernetes#65646](https://github.com/kubernetes/kubernetes/issues/65646)
+- kubectl get events should give a timeline of events [kubernetes/kubernetes#36304](https://github.com/kubernetes/kubernetes/issues/36304)
+
+### Goals
+
+- Add an experimental `events` sub-command under the kubectl and move the existing `events` command from `kubectl get` 
+- This new command will not be dependent on `kubectl get`
+- Help addressing the existing issues which cannot be achieved with `kubectl get events`
+
+## Proposal
+
+Have a independent *events* command which can perform all the existing tasks that the current `kubectl get events` 
+command is performing and also to extend the `kubectl get events` functionality to address the existing issues. 
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+Once the kubectl events command is implemented, this can be rolled out in multiple phases: 
+-  Release the new command `kubectl evetns` alongside with the existing command. 
```

Comment:
events

Reply:
I'm missing releasing this command as experimental, first. This will allow us to improve the command while gathering feedback. Only after the initial phase we'll graduate the command.